shell/completion.bash

```diff
@@ -237,8 +237,10 @@ _fzf_complete() {
   cur="${COMP_WORDS[COMP_CWORD]}"
   if [[ "$cur" == *"$trigger" ]]; then
     cur=${cur:0:${#cur}-${#trigger}}
+    [ -z "${prefix:-}" ] && prefix="${cur}"
+    prefix=${prefix%% }
 
-    selected=$(FZF_DEFAULT_OPTS="--height ${FZF_TMUX_HEIGHT:-40%} --reverse --bind=ctrl-z:ignore ${FZF_DEFAULT_OPTS-} ${FZF_COMPLETION_OPTS-} $str_arg" __fzf_comprun "${rest[0]}" "${args[@]}" -q "$cur" | $post | tr '\n' ' ')
+    selected=$(FZF_DEFAULT_OPTS="--height ${FZF_TMUX_HEIGHT:-40%} --reverse --bind=ctrl-z:ignore ${FZF_DEFAULT_OPTS-} ${FZF_COMPLETION_OPTS-} $str_arg" __fzf_comprun "${rest[0]}" "${args[@]}" -q "$prefix" | $post | tr '\n' ' ')
```

Comment:
Not sure I follow the logic here. Can you explain why you introduced `$prefix`?

Reply:
Bash version doesn't seem to work as advertised.

<img width="405" alt="image" src="https://github.com/junegunn/fzf/assets/700826/2c0a0b97-590d-4618-b296-8d7716e392d4">

Zsh version does work, but I'm getting warning messages.

<img width="650" alt="image" src="https://github.com/junegunn/fzf/assets/700826/451535cf-3080-4a04-b76a-cc1d9ac7fd51">

Reply:
Also, `ssh -i **<tab>` correctly brings up a list of files, but `ssh -i**<tab>` doesn't.

Reply:
That's weird, your bash completion demonstrates prior behavior. Maybe old version of shell/completion.bash got sourced in your terminal after the new one somehow? Can you check, and if it's not the case provide more details about your bash setup? (list of options with `echo $-` would be great)
![bash-test](https://github.com/junegunn/fzf/assets/12125537/7f4a46ce-3ac6-4086-b918-54c86c4802eb)

As for the `ssh -i**<tab>` for me this looks like I'm trying to find completion for a flag, not for an argument, so I didn't really thought about handling this case. If you think this is a valid use-case I can add some logic to check, if the word being completed is a flag or not and then handle it the same way as `ssh -i **<tab>`.

I also was thinking about putting some tests for the ssh completion in the `test/test_go.rb` but for that I would need to add an option to specify additional custom path for `known_hosts`. What do you think?

By the way,

Reply:
I don't understand how it's working on your machine, because if I'm not mistaken, bash doesn't support non-greedy quantifiers which you used in line 305.

```sh
jg@:~/github/fzf (rebrendov/master)> echo $BASH_VERSION
5.1.16(1)-release
jg@:~/github/fzf (rebrendov/master)> echo $-
himBHs
jg@:~/github/fzf (rebrendov/master)> [[ foobar =~ fo+?b ]] && echo 1 || echo nope
nope
jg@:~/github/fzf (rebrendov/master)> [[ foobar =~ fo+b ]] && echo 1 || echo nope
1
```

Reply:
I'm not sure why it worked on my systems, but from what I found in docs it seems like bash uses regex implementation from libc. So the reason of working on my end and not working on yours might be just a different versions of libc (for me it's `ldd (Ubuntu GLIBC 2.35-0ubuntu3.1) 2.35`, `ldd (GNU libc) 2.38` and `ldd (Ubuntu GLIBC 2.31-0ubuntu9.1) 2.31`).

To avoid inconsistencies I got rid of matching groups entirely left just `=~ '@'` check.

Reply:
> I also was thinking about putting some tests for the ssh completion in the test/test_go.rb but for that I would need to add an option to specify additional custom path for known_hosts for ssh_completion. That would solve #2366.

You know I'm hesitant to add options, and I need more careful consideration before making the decision. Having said that, I imagine you can still write a set of tests for the new behavior because most systems should have `127.0.0.1` in the list.

Reply:
> To avoid inconsistencies I got rid of matching groups entirely left just `=~ '@'` check.

Thanks, it's working as expected now. I'll review the code in more detail and see if there's any room for improvement.